File: app/src/main/java/com/kent/university/privelt/ui/data/DataAdapter.java
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.kent.university.privelt.ui.data;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kent.university.privelt.R;
import com.kent.university.privelt.model.UserData;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class DataAdapter extends RecyclerView.Adapter<DataViewHolder> {

    private List<UserData> userData;

    DataAdapter(ArrayList<UserData> userData) {
        this.userData = userData;
    }

    @NonNull
    @Override
    public DataViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.cell_userdata, parent, false);
        return new DataViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DataViewHolder holder, int position) {
        holder.bind(userData.get(position));
    }

    @Override
    public int getItemCount() {
        return userData.size();
    }

    void setUserData(List<UserData> userData) {
        this.userData = userData;
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/data/DataAdapter.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.data

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kent.university.privelt.R
import com.kent.university.privelt.model.UserData
import java.util.*

class DataAdapter internal constructor(userData: ArrayList<UserData>) : RecyclerView.Adapter<DataViewHolder>() {
    private var userData: List<UserData>

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DataViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.cell_userdata, parent, false)
        return DataViewHolder(view)
    }

    override fun onBindViewHolder(holder: DataViewHolder, position: Int) {
        holder.bind(userData[position])
    }

    override fun getItemCount(): Int {
        return userData.size
    }

    fun setUserData(userData: List<UserData>) {
        this.userData = userData
    }

    init {
        this.userData = userData
    }
}